Index: frames.py
===================================================================
--- frames.py	(revision 3035)
+++ frames.py	(working copy)
@@ -1552,6 +1552,8 @@
           while ch != "\x00":
               self.mimeType += ch;
               ch = input.read(1);
+              if len(ch) == 0:
+                 raise FrameException("GEOB frame does not contain a valid mime type");
       else:
           # v2.2 (OBSOLETE) special case
           self.mimeType = input.read(3);
